### ISSUE TITLE
[REQUEST] Implement station usage verification (GTFS rule)

### ISSUE THREAD

**Issue by maximearmstrong** [MobilityData/gtfs-validator#153]
**Is your feature request related to a problem? Please describe.**
A stop and station should be used at least once. This is a GTFS rule implemented in Google Python validator and featured in Google Type Error as TYPE_STATION_UNUSED.

**Describe the solution you'd like**
Actual Google GTFS validator behaviour : verifies a stop from database. Verifies if stop is used, station associated, and if stop is too far from parent station

**Describe alternatives you've considered**

**Additional context**
Line 87 in Error support priorities https://docs.google.com/spreadsheets/d/1vqe6wq7ctqk1EhYkgtZ0_TbcQ91vccfs2daSjn20BLE/edit#gid=0

**Comment by isabelle-dr** [comment]
This is similar to the [unused_shape](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#unused_shape) and [unused_trip](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#unused_trip) validation rules.
Although this is not explicitly mentioned in the spec or best practices, this could be added to this validator if the community sees value in having this check.

**Comment by bradyhunsaker** [comment]
This request was made in 2020.
In 2021 a change was made to warn if any stops do not appear in stop_times:
https://github.com/MobilityData/gtfs-validator/pull/960

That seems to be the most important part.

There seem to be two other parts to this request:
* Verifying that each station is used by at least one stop/platform.
* Verifying that stops/platforms are not too far from their stations.

I don't see either of those in the set of rules.
Are those two desired?

**Comment by derhuerst** [comment]
> * Verifying that each station is used by at least one stop/platform.
> * Verifying that stops/platforms are not too far from their stations.
> 
> […] Are those two desired?

Yes! 🙋

**Comment by isabelle-dr** [comment]
I recommend having INFO for this notice since having stations used isn't an official requirement or recommendation from the spec or Best Practices. 

Opened issue #1348 to follow up about the severity of three other issues that might also need to be downgraded (or to have the spec amended).

**Comment by bradyhunsaker** [comment]
I made PR #1355 to check that a station is used by some stop.

The last part of this issue is checking distance between a stop and its parent_station.
I don't see any guidelines in the best practices document.

There are some large train stations that have platforms pretty far from one another.

Here are some possible approaches:
*  Use a simple threshold. If we accept INFO messages for large train stations, this could be 100 or 200 meters. If we don't want many INFO messages for large train stations, then maybe 500 meters.
*  Use a threshold based on the number of stops. For a two-stop station like a bus stop with stops on opposite sides of a street, use a small threshold like 100 meters (used for shapes and stops). For a many-stop station be more lenient.
*  Use some more complicated logic, like whether the stops in a station form a reasonable cluster (are there outliers?) and whether the station location is central in that cluster.

Anyone have opinions?

**Comment by isabelle-dr** [comment]
Hello @bradyhunsaker, thank you for working on this and sorry for the late answer on this issue.

> The last part of this issue is checking the distance between a stop and its parent_station.

I believe this is part of issue #154, and there is a proposition for what thresholds to use.
Your work on #1355 would close this issue. 🚀

**Comment by emmambd** [comment]
@isabelle-dr Did https://gtfs-validator.mobilitydata.org/rules.html#platform_without_parent_station close out this issue?

**Comment by isabelle-dr** [comment]
No, the two checks mentioned [in this comment](https://github.com/MobilityData/gtfs-validator/issues/153#issuecomment-1449260275) are still relevant. 
They are part of the Python validator but not this one, and they are both out of spec. 
I would recommend adding them as INFO.

Here is the logic for these new rules:

**Verifying that each station is used by at least one stop/platform.**

```
For each stop_id that has location_type set to 1 (a station):
   If the stop_id is not referenced in any of the stops.parent_station:
         Generate an unused_station notice.
```
This is along the lines of he [unused_shape](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#unused_shape) and [unused_trip](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#unused_trip).

**Verifying that stops/platforms are not too far from their stations.**
```
For each stop_id that has location_type set to 1 (a station):
    For each stop_id that has this particular station defined in parent_station:
        Calculate the distance between this stop_id and the station. 
              If the distance is bigger than [threshold], generate a stop_far_from_station notice
```

**Comment by isabelle-dr** [comment]
I just noticed we now have [unused_parent_station](https://gtfs-validator.mobilitydata.org/rules.html#unused_parent_station-rule, so this first one is solved.

**Comment by cswilson252** [comment]
I'd like to start working on the `stop_far_from_station`, just wondering how to calculate a good threshold value for this..

**Comment by emmambd** [comment]
@skalexch  Could you share the threshold for a stop too far from station from the original validator?

**Comment by skalexch** [comment]
@emmambd @cswilson252 the python validator had **100m** to issue a warning and **1km** to issue an error. @emmambd are we okay with having this as an error? 1km is definitely a lot but there might be a case where that limit is stretched. But I am good (as a starter) with 100m to trigger a warning, the acceptance tests will be the judge eventually.

**Comment by emmambd** [comment]
@skalexch  I'd suggest lowering the severity to warning and info, so 1km is a warning and 100m is an info notice.

Hope this helps @cswilson252!

**Comment by cswilson252** [comment]
Great, thanks, I'll start working on this soon! :)
